src/callbacks_step/glm_speed.jl

```diff
@@ -69,17 +74,42 @@ end
 @inline function (glm_speed_callback::GlmSpeedCallback)(integrator)
     dt = get_proposed_dt(integrator)
     semi = integrator.p
-    mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
-    @unpack glm_scale, cfl = glm_speed_callback
 
-    # compute time step for GLM linear advection equation with c_h=1 (redone due to the possible AMR)
-    c_h_deltat = calc_dt_for_cleaning_speed(cfl, mesh, equations, solver, cache)
+    @unpack glm_scale, cfl, semi_indices = glm_speed_callback
 
-    # c_h is proportional to its own time step divided by the complete MHD time step
-    equations.c_h = glm_scale * c_h_deltat / dt
+    if (typeof(semi) <: SemidiscretizationCoupled) && (length(semi_indices) == 0)
+        throw("Since you have more than one semidiscretization you need to specify the 'semi_indices' for which the GLM speed needs to be calculated.")
+    end
+
+    # Make sure we can handle multiple semidiscretizationis in coupled simulations.
+    if length(semi_indices) == 0
+        semis = tuple(semi)
+        semi_indices = (1)
+    else
+        semis = semi.semis
+    end
+
+    # Check that all MHD semidiscretizations received a GLM cleaning speed update.
+    for (semi_index, semi) in enumerate(semis)
+        if (typeof(semi.equations) <: AbstractIdealGlmMhdEquations &&
+            !(semi_index in semi_indices))
+            throw("Equation of semidiscretization $semi_index needs to be included in 'semi_indices' of 'GlmSpeedCallback'.")
+        end
+    end
+
+    for semi_index in semi_indices
+        semi = semis[semi_index]
+        mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+
+        # compute time step for GLM linear advection equation with c_h=1 (redone due to the possible AMR)
+        c_h_deltat = calc_dt_for_cleaning_speed(cfl, mesh, equations, solver, cache)
 
-    # avoid re-evaluating possible FSAL stages
-    u_modified!(integrator, false)
+        # c_h is proportional to its own time step divided by the complete MHD time step
+        equations.c_h = glm_scale * c_h_deltat / dt
+
+        # avoid re-evaluating possible FSAL stages
+        u_modified!(integrator, false)
+    end
```

Comment:
```suggestion
    end

    # avoid re-evaluating possible FSAL stages
    u_modified!(integrator, false)
```
IMHO that needs to be executed only once